src/flutter/shell/platform/linux_embedded/window/linuxes_window_drm.cc

```diff
@@ -94,7 +89,7 @@ bool LinuxesWindowDrm::IsValid() const {
 }
 
 bool LinuxesWindowDrm::DispatchEvent() {
-  // do nothing.
+  sd_event_run(libinput_event_loop_, -1);
```

Comment:
Is `-1` appropriate? `-1` means an infinite timeout.

Reply:
`-1` is not appropriate.
I would like to change it to 10 (microseconds), is that okay?
I have verified x64 Ubuntu and Raspberry Pi4 work with this change.

Reply:
> I would like to change it to 10 (microseconds), is that okay?

Can you investigate a similar code to change it to the appropriate value?

Reply:
[flutter-pi](https://github.com/ardera/flutter-pi) software runs the Flutter application on the DRM backend.
The implementation method is different, but the waiting time seems to be set to "0".
https://github.com/ardera/flutter-pi/blob/9a9e09945a6876b702e0b43f0e143c91f45651ca/src/flutter-pi.c#L937

Since the timing when sd_event comes is not decided, I think there is no problem even if the waiting time is "0".
I have verified x64 Ubuntu and Raspberry Pi4 work with `sd_event_run(libinput_event_loop_, 0);`

Reply:
`sd_event_wait` and `sd_event_run` are different APIs. How does `sd_event_run` work? It dandles the events in event queue? You need to investigate other codes and explain how `sd_event_run` works.

Reply:
I investigated other codes but couldn't find a good sample.

sd_event_run() works as follows:
1. Check the queue with sd_event_prepare().
2-1. If an event exists in the queue, dispatch the event with sd_event_dispatch().
2-2. If no event exists in the queue, wait for the specified time with sd_event_wait().
2-2-1. If an event occurs while waiting, dispatch the event with sd_event_dispatch().
2-2-2. If no event occurs while waiting, nothing to do.

The wait time specified by sd_event_wait() is the value specified by the second argument of sd_event_run (). Therefore, regarding the waiting time value, sd_event_run (xx, 0) and sd_event_wait (xx, 0) have the same settings. However, since [flutter-pi](https://github.com/ardera/flutter-pi/blob/9a9e09945a6876b702e0b43f0e143c91f45651ca/src/flutter-pi.c#L937) does not use sd_event_run(), the processing other than the waiting time does not match. (The main processing flow seems to be the same, but the details are likely to be different)

References:
http://www.devdoc.net/linux/man7.org-20170728/man3/sd_event_prepare.3.html
https://github.com/systemd/systemd/blob/17e90001643d07a7353ba88417e12be2ff0fa042/src/libsystemd/sd-event/sd-event.c#L4094

Reply:
OK. Can you fix the code?

Reply:
I fixed the code. Please re-review.